### PR TITLE
Initial working commit with I2C only

### PR DESCRIPTION
**Pull request by siddacious**
This is to address https://github.com/adafruit/circuitpython/issues/1198

I did notice one issue with the RTD build where the docs for the constructor doesn't understand how to get the value of a const I'm using for a default value and instead displays 
`class adafruit_adxl34x.ADXL345(i2c, address=<sphinx.ext.autodoc.importer._MockObject object>)` instead. If anyone knows how to fix this, please let me know. Making the variable in question not a `const` seems to work but I'd rather fix the docs than change the code.

On thing I was thinking about was the enum-like classes for the `range` and `data_rate` values. Having to say `adafruit_adxl34x.Range.RANGE_2_G` seems repetitive but I can't just have the class variable named `2G`. If anyone can think of a cleaner way to do this, I'm open to it.

### CODE REVIEW

**Comment by microbuilder** [comment]
Thanks for the PR!  

I'm doing some code tests on this today and will do a review after the initial HW sanity checks. I did notice that the first sample is always 0.0 0.0 0.0, though. 

The results being returned are most likely the previous buffered conversion request, so you might simply need a tiny delay between the read request and reading the device output the first time here (between the write that sends the request, and the read to retrieve it): https://github.com/adafruit/Adafruit_CircuitPython_ADXL34x/pull/1/commits/c6c3a3d3764a90ce332979d21c8de7ebc1263138#diff-f32cde418fbcd2f8d3b666d82e153a17R203

File: adafruit_adxl34x.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    - ``DataRate.RATE_6_25HZ``
+    - ``DataRate.RATE_3_13_HZ``
+    - ``DataRate.RATE_1_56_HZ``
+    - ``DataRate.RATE_0_78_HZ``
+    - ``DataRate.RATE_0_39_HZ``
+    - ``DataRate.RATE_0_20_HZ``
+    - ``DataRate.RATE_0_10_HZ``
+
+    """
+    RATE_3200_HZ    = const(0b1111) # 1600Hz Bandwidth   140�A IDD
```
[the remark this answers is not in the document]
Comment (siddacious) [reply adafruit_adxl34x.py]: Yea, I thought it was just my terminal not rendering properly but I'll fix this.

**Comment by siddacious** [comment]
I committed, rebased and force pushed an update that corrects the things you mentioned but my local python environment is pretty fubared and I can't get pylint running even though it's installed. It _should_ be passing pylint now but I'll check again when I get my environment sorted.

**Comment by siddacious** [comment]
D'oh! I recycled the hidden files because the cookiecutter didn't generate them (or I missed something). Should be fixed now.

**Comment by microbuilder** [comment]
I still see a lot of warnings here, mostly about white space (image below), but these are from a different parser I'm using that checks against PEP8 style inside Atom.

pylint seems happy, though, which should be enough for travis.ci.

@kattni Are there specific requirements around PEP8, line spacing, etc. before this can be merged?

screenshot 2018-10-24 at 14 30 28

**Comment by kattni** [comment]
@microbuilder So, in theory, if Pylint passes,  we're happy with it. Personally I'd want some of those fixed up. Your linter seems to be ignoring the `pylint: disable=` line. When it comes to setting registers, the decision to have them all be even with "bad" whitespace is up to the author.

In the end though, as long as Travis is happy, the builds will succeed and that's the important part.

File: adafruit_adxl34x.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+_ADXL345_REG_DATA_FORMAT         = const(0x31) # Data format control
+_ADXL345_REG_DATAX0              = const(0x32) # X-axis data 0
+_ADXL345_REG_DATAX1              = const(0x33) # X-axis data 1
+_ADXL345_REG_DATAY0              = const(0x34) # Y-axis data 0
+_ADXL345_REG_DATAY1              = const(0x35) # Y-axis data 1
+_ADXL345_REG_DATAZ0              = const(0x36) # Z-axis data 0
+_ADXL345_REG_DATAZ1              = const(0x37) # Z-axis data 1
+_ADXL345_REG_FIFO_CTL            = const(0x38) # FIFO control
+_ADXL345_REG_FIFO_STATUS         = const(0x39) # FIFO status
+
```
Comment (caternuson) [inline adafruit_adxl34x.py]: Add a `# pylint: enable=bad-whitespace` here to turn that check back on since you disabled it "globally" up above.

Comment (siddacious) [reply adafruit_adxl34x.py]: done

**Comment by siddacious** [comment]
@caternuson I re-enabled whitespace complaints a bit lower than you requested because there were some additional lines that I wanted it disabled on. Travis looks to be passing again.

**Comment by caternuson** [comment]
I approved, but now realize a better way would be to add the enable where I said. For turning it off in your `Range` class, just add it to the class level modifier.
```python
class Range: #pylint: disable=too-few-public-methods, bad-whitespace
```

**Comment by microbuilder** [comment]
@caternuson I merged this in just to get something started, but feel free to raise an issue with any new suggestions?

**Comment by caternuson** [comment]
@microbuilder Thanks. Sounds good. It's mainly cosmetic linting issues at this point. Not worth holding up a release.

**Pull request merged**